homeassistant/components/climate/zwave.py

```diff
@@ -237,10 +238,23 @@ def set_temperature(self, **kwargs):
             temperature = kwargs.get(ATTR_TEMPERATURE)
         else:
             return
+        operation_mode = kwargs.get(ATTR_OPERATION_MODE)
+        _LOGGER.debug("set_temperature operation_mode=%s", operation_mode)
 
         for value in (self._node.get_values(
                 class_id=zwave.const.COMMAND_CLASS_THERMOSTAT_SETPOINT)
                       .values()):
+            _LOGGER.debug("Testing for operation_mode")
+            if operation_mode is not None:
+                _LOGGER.debug("operation_mode present")
+                setpoint_mode = SET_TEMP_TO_INDEX.get(operation_mode)
+                if value.index != setpoint_mode:
+                    continue
+                _LOGGER.debug("setpoint_mode=%s", setpoint_mode)
+                value.data = temperature
+                break
+
+            _LOGGER.debug("operation_mode not present")
```

Comment:
It is totally possible that both `operation_mode present` and `operation_mode not present` will both be logged as you don't return when an operation_mode is present.